packages/mobile/src/verify/saga.ts

```diff
@@ -822,35 +845,64 @@ export function* getPhoneHashDetails() {
   }
 }
 
+function errorHandler(saga: any, sagaName: string) {
+  return function* (...args: any[]) {
+    try {
+      yield call(saga, ...args)
+    } catch (error) {
+      let errorString = error
+      let lastError = error
+      if (Array.isArray(error)) {
+        errorString = error.join(' - ')
+        lastError = error[-1]
+      }
+      Logger.error(TAG, `Error while running saga ${sagaName}`, error)
+      if (lastError?.toString() === FetchErrorTypes.QuotaExceededError) {
+        yield put(resetKomenciSession())
+      } else {
+        yield put(fail(errorString))
```

Comment:
Does this display a message to the user?

Reply:
Yes, calling `fail` sets the verification state to `Failed` and shows the 'Confirmation Failed' modal with a 'Skip for now' CTA

Reply:
Good, and it's always a user friendly error, right? not the error string we here?

Reply:
Yes, the error string is sent to analytics, the text of the modal is generic and fixed